### ISSUE TITLE
Save State loading breaks MGS Peace Walker missions

### ISSUE THREAD

**Issue by SilverEye91** [RPCS3/rpcs3#15693]
### Quick summary

Saving and loading seems fine on first inspection, but completing missions after loading a save state causes the game to freeze. 

### Details

Saving and loading save states seems to work perfectly on first glance. However, once a mission is completed the ending cutscene plays and the game starts to load returning to Mother Base. At this point the peace walker loading logo spins for a short while before freezing the game entirely. 

I tried it with a couple of the Armor Vehicle and Tank Missions, with the latest one I tried being Extra Op 100 Armor Vehicle Battle: BTR-60 PA Custom. 

I first tested and thus encountered the bug a few days ago and I also redid the test just before submitting this bug report. That was done on the latest version (0.0.32-16594) and here's how I replicated the issue: 

1. I loaded the game normally, from the regular save file.
2. started extra op 100
3. once I neared the end of the fight I created a save state and completed the mission. 

### Attach a log file

[RPCS3.log.gz](https://github.com/user-attachments/files/15791550/RPCS3.log.gz)

### System configuration

Windows 11 Home 10.0.22631 Build 22631
AMD Ryzen 7 5800X
RTX 3070 on 555.99 Game ready drivers
32 GB DDR4 RAM
m.2 PCIe NVMe Gen 4 x4 SSD, Samsung 980 Pro
Gigabyte B550 AORUS ELITE V2

**Comment by digant73** [comment]
@SilverEye91 is that still an issue?

**Comment by elad335** [comment]
Is it working though?

**Comment by SilverEye91** [comment]
Hi! Sorry for not responding before now. I don't know if this is still an issue, I don't play the game anymore so haven't tried it or had the need to do so. This was in the early days of save states on the emulator so I would say that there's a high change that it is fixed. If you want to keep the issue open a few days longer I can try it tomorrow after work.

**Comment by digant73** [comment]
@SilverEye91 if possible, I would say it would be better if you can retest and provide an update

**Comment by SilverEye91** [comment]
@digant73 Alright, I will do that tomorrow then!

**Comment by SilverEye91** [comment]
@digant73 Update:

I've retested the issue, using the same mission as I reported initially on. I did a few quick saves during the mission, as well as loaded it twice during the mission, and once near the end as per my reproduction instructions last time. 

This worked! The system is the same, albeit updated with a 3090 and 128GB of DDR4 RAM. And the install of RPCS3 is the same, but of course periodically updated, the last time today when I opened it to test the issue. 

I can't say that it's completely gone, but I can say that when I ran into this issue two years ago it happened in this game every time I tried save states, and now it didn't. Presumably this is because of the extensive updates that have happened to save states since they were implemented, which I believe was not too far before my initial report. 


TL;DR. It does not appear to be an issue anymore. You can close the ticket. :)

**Comment by digant73** [comment]
thanks for the update. Please close the ticket
